admin/lib/sessions/add.js

```diff
@@ -0,0 +1,41 @@
+module.exports = add
+
+var _ = require('lodash')
+// var accountsFind = require('../accounts/find')
+var accountsFindAll = require('../accounts/find-all')
+var internals = {}
+internals.request = require('../../../utils/request')
+internals.deserialise = require('../../../utils/deserialise')
+internals.serialise = require('../../../utils/serialise')
+
+function add (state, options) {
+  // TODO: use accountsFind 而不是 accountsFindAll
+  //       after updating accountsFind to match admin README doc
+  // return accountsFind(state, {username: options.username})
+  return accountsFindAll(state)
+    .then(function (response) {
+      if (!options || !options.username) {
+        throw new Error('options.username is required')
+      }
```

Comment:
move the check before the find call. You can do `return Promise.reject(new Error('options.username is required'))` instead of the `throw`